### PR TITLE
Tree search

### LINKED ISSUE #2632: Browser search (ctrl+F) is limited to visible part of tree list
https://discuss.webknossos.org/t/browser-search-ctrl-f-is-limited-to-visible-part-of-tree-list/894

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/2632/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by philippotto**
### Mailable description of changes:
 - Trees can be searched via a search menu within the trees tab on the right hand side. In contrast to the browser search functionality (CTRL + F) this search will consider **all** trees regardless of whether they are visible in the viewport.
- The comments tab will load significantly faster for a large amount of trees.

image

### Steps to test:
- Open a tracing with trees and use the search functionality in the tree tabs view
- Clicking a selected tree (or hitting enter) should close the search and focus the tree in the list.

### Notes
I tested the search feature for a tracing with 10 000 trees and the performance was near-instant. While testing that I noticed that the comments tab took 7 seconds to load. I fixed that by pulling a filter in front of a sort, since I measured that the sorting was the bottleneck. Now, the comment tab performance loads near-instant (for zero comments).

### Issues:
- fixes #2632 

------
- [X] Ready for review

### CODE REVIEW

File: app/assets/javascripts/oxalis/view/right-menu/tree_search_popover.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                  <Option key={tree.treeId} text={tree.name}>
+                    {tree.name}
+                  </Option>
+                ))}
+              style={{ width: "500px" }}
+              ref={autoComplete => {
+                this.autoComplete = autoComplete;
+              }}
+              onSelect={(value, option) => {
+                console.log("onSelect", value, option);
```
Comment (daniel-wer) [inline app/assets/javascripts/oxalis/view/right-menu/tree_search_popover.js]: remove console.log

**Comment by philippotto** [comment]
@daniel-wer I added `ctrl + shift + f` as a shortcut and `escape` will close the popover again. As we agreed on, I didn't intercept `ctrl + f` since this would be annoying for the comments tab. I think the search button should be prominent enough.

Feel free to review my last two commits. I added a react component which abstracts from the input library so that the shortcut hooks are only active when necessary.

File: app/assets/javascripts/libs/input.js [hunk 2]
```diff
@@ -120,9 +127,16 @@ export class InputKeyboard {
   bindings: Array<KeyboardBindingDownUp> = [];
   isStarted: boolean = true;
   delay: number = 0;
+  supportInputElements: boolean;
```
Comment (daniel-wer) [inline app/assets/javascripts/libs/input.js]: This is missing the default value of false I think.

**Comment by daniel-wer** [comment]
@philippotto Works beautifully, feel free to merge :)

File: app/assets/javascripts/libs/input.js (hunk 2 above)
Comment (philippotto) [reply app/assets/javascripts/libs/input.js]: Nice catch! Thanks!

**Pull request merged**